updatecli/updatecli.d/updatecli.yml

```diff
@@ -0,0 +1,69 @@
+---
+name: "Bump Go version in updatecli"
+scms:
+  kine:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ requiredEnv .github.username }}"
+      token: '{{ requiredEnv .github.token }}'
+      owner: "{{ .kine.org }}"
+      repository: "{{ .kine.repo }}"
+      branch: "{{ .kine.branch }}"
+  go:
+    kind: "github"
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      username: "{{ requiredEnv .github.username }}"
+      token: '{{ requiredEnv .github.token }}'
+      owner: "{{ .go.org }}"
+      repository: "{{ .go.repo }}"
+      branch: "{{ .go.branch }}"
+
+sources:
+  latestGoTag:
+    name: "Get latest Go 1.x tag"
+    kind: "gittag"
+    scmid: "go"
+    spec:
+      versionfilter:
+        kind: "regex"
+        pattern: 'go1\.[2-9][0-9]*\.\d$' # >= go1.20 < go2
```

Comment:
One quirk in the system is that unless your source uses proper semver then the sorting will be based on historical timeline rather than the tag numbers. The golang/go repo doesn't use proper semver (see the 'go' in front of the tag name), so it is not sorted alphabetically, but historically. The go1.19.7 tag was pushed after the go1.20.2 tag (the repo doesn't use GH releases), so the go1.19.7 tag was coming up as "latest", this regex makes sure that 1.20+ is selected, but we could run into the same issue when go1.30 is tagged.

Reply:
yes, I ran into this error as well and found that I needed to use the old un-scoped GH token style, which I assumed was being used on the runners. I don't have any insight into that environment, so thank you for the additional context! 
As for what to do about it... I guess having updatecli update its GH action file is not going to be viable, I will remove this file for now.

Reply:
Yes, unless K3s team has a bot account for such tasks.

Reply:
I would honestly like that GH could solve this issue. A bunch of people has the same problem and have to use one of those workarounds. 😮‍💨